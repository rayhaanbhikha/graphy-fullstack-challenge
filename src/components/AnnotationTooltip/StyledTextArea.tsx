import styled from 'styled-components';

export interface IStyledTextArea {
  inEditMode: boolean;
}

export const StyledTextArea = styled.textarea.attrs<IStyledTextArea>(props => ({
  readonly: !props.inEditMode,
  disabled: !props.inEditMode,
})) <IStyledTextArea>`
  flex-grow: 3;
  min-width: inherit;
  border: none;
  background: ${props => props.inEditMode ? 'white' : 'inherit'};
  margin: 2px;
  color: black;
  cursor: ${props => props.inEditMode ? 'text' : 'inherit'}
`;