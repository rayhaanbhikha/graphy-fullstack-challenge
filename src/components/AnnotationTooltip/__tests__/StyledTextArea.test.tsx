import React from 'react';
import { render } from '@testing-library/react'
import { StyledTextArea } from '../StyledTextArea';
import 'jest-styled-components';

describe('StyledTextArea', () => {

  it('should have correct default styles', () => {
    const { asFragment } = render(<StyledTextArea inEditMode={false} />);
    expect(asFragment()).toMatchSnapshot();
  })

  it('should have correct styles when in edit state', () => {
    const { asFragment } = render(<StyledTextArea inEditMode={true} />);
    expect(asFragment()).toMatchSnapshot();
  })
});