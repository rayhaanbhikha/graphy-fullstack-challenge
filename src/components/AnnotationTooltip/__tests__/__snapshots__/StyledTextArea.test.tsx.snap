// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledTextArea should have correct default styles 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-box-flex: 3;
  -webkit-flex-grow: 3;
  -ms-flex-positive: 3;
  flex-grow: 3;
  min-width: inherit;
  border: none;
  background: inherit;
  margin: 2px;
  color: black;
  cursor: inherit;
}

<textarea
    class="c0"
    disabled=""
  />
</DocumentFragment>
`;

exports[`StyledTextArea should have correct styles when in edit state 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-box-flex: 3;
  -webkit-flex-grow: 3;
  -ms-flex-positive: 3;
  flex-grow: 3;
  min-width: inherit;
  border: none;
  background: white;
  margin: 2px;
  color: black;
  cursor: text;
}

<textarea
    class="c0"
  />
</DocumentFragment>
`;
